_posts/2022-04-12-identity.markdown

```diff
@@ -0,0 +1,76 @@
+---
+layout: post
+title:  "How Flotta project proofs identity on the far edge"
+date:   2022-04-12 12:00:00 +0100
+categories: flotta
+author: Eloy Coto <eloy.coto@acalustra.com>
+tags:
+  - authentication
+  - flotta
+
+summary-1: How Flotta project proofs identity on the far edge.
+---
+
+When speaking about the edge one of the most problematic things is the security
+point of view. At the edge, you cannot assume there is security at all, so we
+assume that all your devices are compromised by default and we adopt a [Zero
+trust](https://cloud.google.com/beyondcorp) overview.
+
+Edge Authentication is getting traction, and at the moment, there are multiple
+projects out there that it’s taking care of it, for us, the most promised one,
+is the protocol-driven by the [FIDO alliance](https://fidoalliance.org/),
+regarding Device Onboarding, also known as
+[FDO](https://fidoalliance.org/specs/FDO/FIDO-Device-Onboard-RD-v1.0-20201202.html).
+
+Flotta will use FDO for initialization, but regarding edge security, there are
+more steps than that, like Certificate renewals, certificates expiration and
+much more, for that Flotta has some flows in place.
+
+
+## Device Onboarding
+
+Right now, Flotta uses [Mutual
+TLS](https://en.wikipedia.org/wiki/Mutual_authentication) certificates to
+onboard new devices to the Operator. That certificate will be injected into the
+ISO creation using kickstart files where Manufacturer or Edge administrator can
+initialise the device.
+
+When the device boots on the final destination, it’ll use that Register
+certificate to register (The only action that can me with that certificate),
+such API call also contains a Certificate signed Request with a public key to
+be able to sign it and returned to the device. That certificate can only be
+obtained if the Edge-administrator approves the device to register first.
+
+To avoid security issues with that certificates, Flotta will implement TPM keys
+for that certificates, so if someone has access to the certificate cannot be
+used outside of the device. At the same time, in the API a new security check
+will be added that devices can only be registered if the Key is TPM based.
+
+
+## Authentication
+
+When designing the authentication system in Flotta, we know that are some
+special cases, Edge network is not reliable, devices can be turned off for a
+long period of time, and the API Server should scale horizontally. Because of
```

Comment:
Do you think it makes sense to explain why `the API Server should scale horizontally`?